recipe/meta.yaml

```
@@ -22,7 +22,7 @@ requirements:
   run:
     - python
     - deprecated >=1.2.6
-    - opentelemetry-api ~=1.3
+    - opentelemetry-api >=1.3,<2.0
 
```

```suggestion
    - opentelemetry-api >=1.3,<2.0.dev0
```
Or you could just follow upstream syntax and use `~=1.3`. I believe that is valid for conda, right?

---

My understanding was `~=` is not valid, and I couldn't find in the documentation anywhere to convince me otherwise. 

This was GPT's response:

```
No, that syntax (~=) does not work in a conda recipe.

Conda recipes use meta.yaml, which expects version specifiers to follow PEP 440-style syntax only when using pip, but conda itself uses a slightly different syntax. In a conda recipe’s requirements section, valid specifiers include:
	•	Exact version: opentelemetry-api ==1.3
	•	Minimum version: opentelemetry-api >=1.3
	•	Range: opentelemetry-api >=1.3,<2.0
```


I can try rendering a recipe to confirm or deny.

---

When I do a dry run install of `opentelemetry-propagator-b3 v1.26.0` it wants to install `opentelemetry-api-1.26.0-py312hca03da5_0` as a dependency. I assume it's using the same library as pip to resolve version specifiers but I'm not certain. Seems to work, though.

---

The CLI definitely handles it: https://github.com/conda/conda/blob/main/conda/cli/common.py#L138

I _think_ maybe this is where a recipe line would be parsed? https://github.com/conda/conda/blob/main/conda/common/pkg_formats/python.py#L1099

---

Cool. TIL.

---

Or maybe it all falls back to the MatchSpec class (which looks to be what we must have based some of the `ddb get-impact` logic on because it looks very similar)
https://github.com/conda/conda/blob/main/conda/models/match_spec.py#L309